ime/app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardSuggestions.java

```diff
@@ -1236,10 +1236,34 @@ public void pickSuggestionManually(
                     suggestion,
                     suggestion /*user physically picked a word from the suggestions strip. this is not a fix*/);
 
-            // Follow it with a space
+            // Follow it with a space if there is not already one or if it is not a punctuation mark
+            // that goes attached to the word being manually picked
             if (withAutoSpaceEnabled && (index == 0 || !typedWord.isAtTagsSearchState())) {
-                sendKeyChar((char) KeyCodes.SPACE);
-                setSpaceTimeStamp(true);
+                boolean isNextCharSpaceOrPunctuation = false;
+                if (ic != null) {
+                    String strNextChar = ic.getTextAfterCursor(1, 0).toString();
+                    if (strNextChar.length() == 1) {
+                        char nextCharAfterCursor = strNextChar.charAt(0);
+                        isNextCharSpaceOrPunctuation =
+                                nextCharAfterCursor == ' '
```

Comment:
This can be replaced with a `HashSet` as a member field. Would probably be faster to check, and easier to expand if needed.